### DISCUSSION TITLE
Screenshot Screen, get URL and Upload to Webhook

### DISCUSSION THREAD

**Discussion by bflynn631** [iseahound/ImagePut#54] [Q&A] [answered]
Hello! What I am basically looking to do is send a webhook to discord with an image of the users screen at the time of it being sent. Ive tried multiple different things that did not work and then came across ImagePut, which seems like it could be used to achieve this although I don't know how. If anyone could provide some help and/or guidance that would be great, thanks!

The sending of the webhook works already and everything, its just this image addition i need some help with. Also just to clarify this is Autohotkey v2

(Here is a snippet of my code)

```
SendTestWebhook(*) {
    global webhookURL
    url := webhookURL.Value

    if (url = "") {
        MsgBox("Webhook URL is empty. Please enter a valid URL.")
        return
    }

    ; Define the JSON payload for the test message
    postdata := "{"
    postdata .= '"embeds": [{'
    postdata .= '"title": "Test Msg",'
    postdata .= '"description": "This is a test message!",'
    postdata .= '"thumbnail": {"url": "' screenshotURL '"},'    **; SCREENSHOT OF USERS CURRENT SCREEN AT THE TIME OF THE WEBHOOK BEING SENT HERE**
    postdata .= '"color": 16738740,'
    postdata .= '"footer": {"text": "Test Footer", "icon_url": "https://cdn.discordapp.com/attachments/1058152472536940659/1283843219804782633/lgow.png?ex=66e477ac&is=66e3262c&hm=b4f10f2b4329e00f9da4d63be6ac47e4ba92eb45fa898503736cc58c9df511dd&"},'
    postdata .= '"timestamp": "2023-10-13T00:00:00.000Z"'
    postdata .= "}]"
    postdata .= "}"

    ; Send the HTTP POST request
    WebRequest := ComObject("WinHttp.WinHttpRequest.5.1")
    WebRequest.Open("POST", url, false)
    WebRequest.SetRequestHeader("Content-Type", "application/json")
    WebRequest.Send(postdata)

    if (WebRequest.Status = 204) {
        MsgBox("Test message sent successfully.")
    } else {
        MsgBox("Failed to send test message. Status: " WebRequest.Status)
    }
}
```

**Comment by iseahound** [comment]
Hopefully someone else can answer because I'm not familiar with API endpoints like discord

**Reply by bflynn631** [reply]
Well the fact that its going to discord exactly doesn't really matter. 

All i really need to do is find a way to get a screenshot of the users current screen, and convert that to a link so i can then include it in that webhook.

**Reply by iseahound** [reply, marked as answer]
ImagePutURL([x, y, w, h]) or ImagePutURL(0) will return an Imgur url. You may be rate limited since I don't offer the ability to use your own imgur API key.
